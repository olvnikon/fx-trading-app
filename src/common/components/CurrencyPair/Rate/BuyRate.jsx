// @flow
import React from 'react';
import Rate from './Rate';
import styles from './Rate.scss';

type Props = {|
  +currency: string,
  +rate: number,
|};

const BuyRate = ({ currency, ...rest }: Props) => (
  <div className={styles.buyRate}>
    <Rate header={`Buy ${currency}`} {...rest} />
  </div>
);

export default BuyRate;
