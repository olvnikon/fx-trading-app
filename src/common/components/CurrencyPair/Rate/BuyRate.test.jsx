// @flow
import React from 'react';
import { shallow } from 'enzyme';
import BuyRate from './BuyRate';

describe('BuyRate', () => {
  const RATE = 12.345;
  const CURRENCY = 'USD';

  test('should match snapshot', () => {
    const props = {
      rate: RATE,
      currency: CURRENCY,
    };
    const component = shallow(<BuyRate {...props} />);
    expect(component).toMatchSnapshot();
  });
});
