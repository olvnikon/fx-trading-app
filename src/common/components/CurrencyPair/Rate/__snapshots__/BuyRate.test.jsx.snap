// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuyRate should match snapshot 1`] = `
<div>
  <Rate
    header="Buy USD"
    rate={12.345}
  />
</div>
`;
